src/backend/cpu/morph.cpp

```diff
@@ -22,14 +23,30 @@ namespace cpu
 template<typename T, bool isDilation>
 Array<T> morph(const Array<T> &in, const Array<T> &mask)
 {
+    af::borderType padType = isDilation ? AF_PAD_ZERO : AF_PAD_CLAMP_TO_EDGE;
+
     in.eval();
     mask.eval();
 
-    Array<T> out = createEmptyArray<T>(in.dims());
+    const af::dim4 idims = in.dims();
+    const af::dim4 mdims = mask.dims();
 
-    getQueue().enqueue(kernel::morph<T, isDilation>, out, in, mask);
+    const af::dim4 lpad(mdims[0]/2, mdims[1]/2, 0, 0);
+    const af::dim4 upad(lpad);
+    const af::dim4 odims(lpad[0] + idims[0] + upad[0],
+                         lpad[1] + idims[1] + upad[1],
+                         idims[2], idims[3]);
 
-    return out;
+    auto out = createEmptyArray<T>(odims);
+    auto inp = padArrayBorders(in, lpad, upad, padType);
+
+    getQueue().enqueue(kernel::morph<T, isDilation>, out, inp, mask);
+
+    std::vector<af_seq> idxs(4, af_span);
```

Comment:
use std::array instead of vector.

Reply:
It can't be done unless createSubArray is changed